src/components/Post/PostContent.tsx

```diff
@@ -0,0 +1,150 @@
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+import { COLOR } from '../../utils/color';
+import { queryLowImage } from '../../utils/image';
+import { ROUTES } from '../../utils/routes';
+import Image from '../Base/Image';
+
+interface PostContentProps {
+  _id: string;
+  title: string;
+  image?: string;
+  isDetailPage?: boolean;
+  isMyLikes?: boolean;
+}
+
+interface PostDetailProps {
+  isDetailPage?: boolean;
+  isMyLikes?: boolean;
+}
+
+interface ImageContainerProps {
+  width?: string;
+  height?: string;
+  isMyLikes?: boolean;
+}
+
+const PostContent = ({
+  _id,
+  title,
+  image,
+  isDetailPage,
+  isMyLikes,
+}: PostContentProps) => {
+  const navigate = useNavigate();
+  const postContent = title[0] === '{' ? JSON.parse(title) : title;
+
+  const toPostDetail = () => {
+    navigate(ROUTES.POSTS_BY_ID(_id));
+  };
+
+  if (isDetailPage) {
+    return (
+      <Section>
+        <Title isDetailPage={isDetailPage}>
+          {typeof postContent === 'string' ? postContent : postContent.title}
+        </Title>
+        {image && (
+          <ImageContainer width="100%">
+            <Image
+              src={queryLowImage(image, 'postDetail')}
+              alt="post-image"
+              objectFit="contain"
+            />
+          </ImageContainer>
+        )}
+        <Text isDetailPage={isDetailPage}>
+          {typeof postContent === 'string' ? postContent : postContent.body}
+        </Text>
+      </Section>
+    );
+  }
+
+  if (isMyLikes) {
+    return (
+      <PostWrapper onClick={toPostDetail}>
+        <Title isMyLikes={isMyLikes}>
+          {typeof postContent === 'string' ? postContent : postContent.title}
+        </Title>
+        <Text>
+          {typeof postContent === 'string' ? postContent : postContent.body}
+        </Text>
+      </PostWrapper>
+    );
+  }
+
+  return (
+    <Section onClick={toPostDetail}>
+      <Title isDetailPage={isDetailPage}>
+        {typeof postContent === 'string' ? postContent : postContent.title}
```

Comment:
```suggestion
        {postContent.title}
```
여기도 위의 코멘트와 같은 맥락입니다!